modules/osint/github_leaks.py

```diff
@@ -0,0 +1,138 @@
+"""
+OWASP Maryam!
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import concurrent.futures
+from bs4 import BeautifulSoup
+import json
+
+meta = {
+	'name': 'GitHub Leaks',
+	'author': 'Aman Singh',
+	'version': '0.1',
+	'description': 'Search your query in the GitHub and show the potentially leaked info.',
+	'sources': ('google', 'carrot2', 'bing', 'yippy', 'yahoo', 'millionshort', 'qwant', 'duckduckgo', 'github'),
+	'options': (
+		('query', None, True, 'Query string', '-q', 'store', str),
+		('limit', 1, False, 'Search limit(number of pages, default=1)', '-l', 'store', int),
+		('count', 50, False, 'Number of links per page(min=10, max=100, default=50)', '-c', 'store', int),
+		('engine', 'google,github', False, 'Engine names for search(default=google, github)', '-e', 'store', str),
+		('thread', 2, False, 'The number of engine that run per round(default=2)', '-t', 'store', int),
+	),
+    'examples': ('github_leaks-q <QUERY> -l 15 -e carrot2,bing,qwant --output',)
+}
+
+LINKS = []
+PAGES = ''
+REPO = []
+OUTPUT = {'links': {}}
+
+def checks(self, repo):
+	checks = ['filename:.npmrc filename:.dockercfg filename:id_rsa filename:credentials filename:.s3cfg',#NOT MORE THAN 500 CHAR IN ONE QUERY
+	'filename:wp-config.php filename:.htpasswd filename:.git-credentials filename:.bashrc filename:.bash_profile', 
+	'filename:.netrc filename:config filename:connections.xml filename:express.conf filename:.pgpass', 
+	'filename:proftpdpasswd filename:server.cfg filename:.bash_history filename:.cshrc filename:.history filename:.sh_history filename:sshd_config', 
+	'filename:dhcpd.conf filename:prod.exs filename:shadow filename:passwd filename:.esmtprc filename:logins.json', 
+	'filename:CCCam.cfg filename:settings filename:secrets filename:master.key filename:WebServers.xml',
+	'password', 'DB_USERNAME', 'TOKEN', 'API_KEY' , 'SECRET_KEY', 'credentials',
```

Comment:
Doesn't it work? `password OR DB_USERNAME OR TOKEN...`

Reply:
No. for eg. https://github.com/saeeddhqan/Maryam/search?q=page shows results but https://github.com/saeeddhqan/Maryam/search?q=page+or+somethingthatdoesn%27texist does not show results.